File: 392. Is Subsequence/solution.go
func isSubsequence(s string, t string) bool {
	p := 0
	ls := len(s)
	if ls == 0 {
		return true
	}
	if t == "" {
		return false
	}
	for i := 0; i < len(t); i++ {
		if s[p] == t[i] {
			p++
			if p >= ls {
				return true
			}
		}
	}
	return false
}

File: 392. Is Subsequence/solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
  for (let i = 0; i < s.length; i++) {
    const index = t.indexOf(s.charAt(i));
    t = t.slice(index + 1);
    if (index == -1) return false;
  }
  return true;
};
